plugins/processors/awsapplicationsignals/internal/resolver/attributesresolver.go

```diff
@@ -0,0 +1,162 @@
+			}
+		}
+	}
+	return &attributesResolver{
+		subResolvers: subResolvers,
+	}
+}
+
+// Process the attributes
+func (r *attributesResolver) Process(attributes, resourceAttributes pcommon.Map, _ bool) error {
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Process(attributes, resourceAttributes); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *attributesResolver) Stop(ctx context.Context) error {
+	var errs error
+	for _, subResolver := range r.subResolvers {
+		if err := subResolver.Stop(ctx); err != nil {
+			errs = errors.Join(errs, err)
+		}
+	}
+	return errs
+}
+
+type resourceAttributesResolver struct {
+	defaultEnvPrefix string
+	platformType     string
+	attributeMap     map[string]string
+}
+
+func newResourceAttributesResolver(defaultEnvPrefix, platformType string, attributeMap map[string]string) *resourceAttributesResolver {
+	return &resourceAttributesResolver{
+		defaultEnvPrefix: defaultEnvPrefix,
+		platformType:     platformType,
+		attributeMap:     attributeMap,
+	}
+}
+func (h *resourceAttributesResolver) Process(attributes, resourceAttributes pcommon.Map) error {
+	for attrKey, mappingKey := range h.attributeMap {
+		if val, ok := resourceAttributes.Get(attrKey); ok {
+			attributes.PutStr(mappingKey, val.Str())
+		}
+	}
+	if _, ok := attributes.Get(attr.AWSLocalEnvironment); !ok {
+		if val, found := resourceAttributes.Get(attr.AWSHostedInEnvironment); found {
+			attributes.PutStr(attr.AWSLocalEnvironment, val.Str())
+		} else {
+			if h.defaultEnvPrefix == appsignalsconfig.PlatformECS {
+				if clusterName, ok := getECSClusterName(resourceAttributes); ok {
+					attributes.PutStr(attr.AWSLocalEnvironment, GetDefaultEnvironment(h.defaultEnvPrefix, clusterName))
+				}
+			}
+			if h.defaultEnvPrefix == appsignalsconfig.PlatformEC2 {
+				if asgAttr, ok := resourceAttributes.Get(attr.ResourceDetectionASG); ok {
+					attributes.PutStr(attr.AWSLocalEnvironment, GetDefaultEnvironment(h.defaultEnvPrefix, asgAttr.Str()))
+				}
+			}
+		}
+	}
+	if _, ok := attributes.Get(attr.AWSLocalEnvironment); !ok {
+		attributes.PutStr(attr.AWSLocalEnvironment, GetDefaultEnvironment(h.defaultEnvPrefix, AttributeEnvironmentDefault))
+	}
+	attributes.PutStr(common.AttributePlatformType, h.platformType)
+
+	return nil
+}
+
+func getECSClusterName(resourceAttributes pcommon.Map) (string, bool) {
+	if clusterAttr, ok := resourceAttributes.Get(semconv1.AttributeAWSECSClusterARN); ok {
+		parts := strings.Split(clusterAttr.Str(), "/")
+		clusterName := parts[len(parts)-1]
+		return clusterName, true
+	} else if taskAttr, ok := resourceAttributes.Get(semconv1.AttributeAWSECSTaskARN); ok {
+		parts := strings.SplitAfterN(taskAttr.Str(), ":task/", 2)
+		if len(parts) == 2 {
+			taskParts := strings.Split(parts[1], "/")
+			// cluster name in ARN
+			if len(taskParts) == 2 {
+				return taskParts[0], true
+			}
+		}
+	}
+	return "", false
+}
+
+func GetDefaultEnvironment(platformCode, val string) string {
```

Comment:
nit: Doesn't need to be exported.
```suggestion
func getDefaultEnvironment(platformCode, val string) string {
```

Reply:
Done.